src/EFCore.MySql/Migrations/MySqlMigrationsSqlGenerator.cs

```diff
@@ -158,15 +158,36 @@ protected override void Generate(RenameIndexOperation operation, IModel model, M
 
             if (operation.NewName != null)
             {
-                builder.Append("ALTER TABLE ")
-                    .Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(operation.Table, operation.Schema))
-                    .Append(" RENAME INDEX ")
-                    .Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(operation.Name))
-                    .Append(" TO ")
-                    .Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(operation.NewName))
-                    .AppendLine(";");
-
-                EndStatement(builder);
+                if (_options.ConnectionSettings.ServerVersion.SupportsRenameIndex)
+                {
+                    builder.Append("ALTER TABLE ")
+                        .Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(operation.Table, operation.Schema))
+                        .Append(" RENAME INDEX ")
+                        .Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(operation.Name))
+                        .Append(" TO ")
+                        .Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(operation.NewName))
+                        .AppendLine(";");
+
+                    EndStatement(builder);
+                }
+                else
+                {
+                    builder.Append("ALTER TABLE ")
+                        .Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(operation.Table, operation.Schema))
+                        .Append(" DROP INDEX ")
+                        .Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(operation.Name))
+                        .AppendLine(";");
+
+                    EndStatement(builder);
+                    
+                    builder.Append("ALTER TABLE ")
+                        .Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(operation.Table, operation.Schema))
+                        .Append(" CREATE INDEX ")
```

Comment:
I just realized this won't work; it doesn't define any column names.  I'm re-working this function so that it can detect `UNIQUE`, `BTREE`, etc. using the `SHOW CREATE TABLE` logic anyways though.